File: comu/com.energyict.mdc.device.data/src/main/java/com/energyict/mdc/device/data/impl/configproperties/ConfigProperties.java
/*
 * Copyright (c) 2019 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.device.data.impl.configproperties;

public enum ConfigProperties {
    TRUE_MINIMIZED("communication.settings.true.minimized"),
    RANDOMIZATION("communication.settings.randomization");

    private String value;
    ConfigProperties(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: coko/com.elster.jupiter.config.extjs/src/main/web/js/cfg/src/properties/model/ConfigProperties.js
/*
 * Copyright (c) 2019 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Cfg.properties.model.ConfigProperties', {
    extend: 'Ext.data.Model',
    requires: [
        'Cfg.properties.model.Properties'
    ],

    fields: [
        {name: 'type', type: 'auto'},
    ],

    associations: [
        {
            name: 'properties',
            type: 'hasMany',
            model: 'Apr.model.TaskProperties',
            associationKey: 'properties',
            foreignKey: 'properties',
            getterName: 'getProperties',
            setterName: 'setProperties',
            getTypeDiscriminator: function (node) {
                return 'Cfg.properties.model.Properties';
            }
        }
    ],

    proxy: {
        type: 'rest',
        url: '/api/mtr/cfgprops',
        reader: {
            type: 'json'
        }
    }
});